R/backend-.R

```diff
@@ -84,6 +84,23 @@ base_scalar <- sql_translator(
     sql_expr(!!x %OR% !!y %AND NOT% (!!x %AND% !!y))
   },
 
+  # bitwise operators
+  # SQL Syntax reference links:
+  #   Hive: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+UDF#LanguageManualUDF-ArithmeticOperators
+  #   Impala: https://www.cloudera.com/documentation/enterprise/5-9-x/topics/impala_bit_functions.html
+  #   PostgreSQL: https://www.postgresql.org/docs/7.4/functions-math.html
+  #   MS SQL: https://docs.microsoft.com/en-us/sql/t-sql/language-elements/bitwise-operators-transact-sql?view=sql-server-2017
+  #   MySQL https://dev.mysql.com/doc/refman/5.7/en/bit-functions.html
+  #   Oracle: https://docs.oracle.com/cd/E19253-01/817-6223/chp-typeopexpr-7/index.html
+  #   SQLite: https://www.tutorialspoint.com/sqlite/sqlite_bitwise_operators.htm
+  #   Teradata: https://docs.teradata.com/reader/1DcoER_KpnGTfgPinRAFUw/h3CS4MuKL1LCMQmnubeSRQ
+  bitwNot    = function(x) sql_expr(~ !!x),
+  bitwAnd    = function(x, y) sql_infix("&")(x, as.integer(y)),
```

Comment:
Shouldn't just this be `bitwAnd = sql_infix("&")`? I don't think you should be coercing the right hand side to an integer.

Reply:
Bitwise operations require that both arguments are integers. My thoughts were:
- If `x` is a column then I don't want to implicitly cast this to an integer (preventing the user from accidentally applying bitwise logic to floating point columns).
- If `y` is a single integer (e.g. `128`) then I want to coerce this to an integer (so the user doesn't need to type `bitwAnd(x, as.integer(128))` or `bitwAnd(x, 128L)`).

Of course, now I realize that we should actually treat both arguments the same. This leaves us with 3 possibilities:
1. Coerce both arguments
2. Coerce neither argument
3. Coerce arguments only if they aren't referring to a column

The base R functions have chosen option 1.

Reply:
I think you should coerce neither, and rely on the database to do it (or otherwise complain).

Reply:
Ok, I've made the change 👍